sympy/tensor/array/arrayop.py

```diff
@@ -296,3 +296,57 @@ def permutedims(expr, perm):
         new_array[i] = expr[t]
 
     return type(expr)(new_array, new_shape)
+
+
+class Flatten(object):
+
+    def __init__(self, iterable):
+        from sympy.matrices.matrices import MatrixBase
+
+        if not isinstance(iterable, (Iterable, MatrixBase)):
+            raise NotImplementedError("Data type not yet supported")
+        if isinstance(iterable, list) and len(iterable)>0 and isinstance(iterable[0], NDimArray):
```

Comment:
if it's a list, then transform it into an `Array`.

Reply:
If we transform it to `Array`, there will be a loop of calling functions:
While initiating an Array, we will call `Flatten()` in return.

Reply:
You cannot add a `list` to a SymPy object. What about calling `flatten( )` in `Array`?

Reply:
OK!